flattentool/json_input.py

```diff
@@ -291,22 +314,40 @@ def parse_json_dict(self, json_dict, sheet, json_key=None, parent_name='', flatt
                     flattened_dict[sheet_key(sheet, parent_name+key)] = ';'.join(map(six.text_type, value))
                 else:
                     if self.rollup and parent_name == '': # Rollup only currently possible to main sheet
+                        
+                        if self.use_titles and not self.schema_parser:
+                            warn('Warning: No schema was provided so column headings are JSON keys, not titles.')
+
                         if len(value) == 1:
                             for k, v in value[0].items():
-                                if self.use_titles and parent_name+key+'/0/'+k in self.schema_parser.main_sheet.titles:
-                                    if type(v) in BASIC_TYPES:
-                                        flattened_dict[sheet_key_title(sheet, parent_name+key+'/0/'+k)] = v
-                                    else:
-                                        raise ValueError('Rolled up values must be basic types')
-                                elif not self.use_titles and parent_name+key+'/0/'+k in self.schema_parser.main_sheet:
-                                    if type(v) in BASIC_TYPES:
+
+                                if self.preserve_fields and parent_name+key+'/'+k not in self.preserve_fields:
+                                    continue
+
+                                if type(v) not in BASIC_TYPES:
+                                    raise ValueError('Rolled up values must be basic types')
+                                else:
+                                    if self.schema_parser:
+                                        if self.use_titles and \
+                                        parent_name+key+'/0/'+k in self.schema_parser.main_sheet.titles:
+                                            flattened_dict[sheet_key_title(sheet, parent_name+key+'/0/'+k)] = v
+                                        elif not self.use_titles and \
+                                        parent_name+key+'/0/'+k in self.schema_parser.main_sheet:
+                                                flattened_dict[sheet_key(sheet, parent_name+key+'/0/'+k)] = v
+                                    elif parent_name+key in self.rollup:
```

Comment:
I think the same applies here.  There could be a schema_parser but if there is no rollup in it I think the command line arguments should be applied.